presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftMetastoreModule.java

```diff
@@ -44,6 +46,7 @@ protected void setup(Binder binder)
     {
         binder.bind(HiveMetastoreClientFactory.class).in(Scopes.SINGLETON);
         binder.bind(HiveCluster.class).to(StaticHiveCluster.class).in(Scopes.SINGLETON);
+        binder.bind(ColumnConverter.class).to(HiveColumnConverter.class).in(Scopes.SINGLETON);
```

Comment:
Let's discuss this in person. Try to understand why we need to bind the class twice in two modules.

Reply:
I stepped through the stack of code to understand what's happening.

We have two binding places (here and at HiveClientModule) because this class is consumed by
1) ThriftHiveMetastore (bounded to HiveMetastore in this module) which uses `fromMetastoreApiTable` that requires us to pass into a column converter
2) HiveMetadataFactory (used in HiveClientModule) which passes the column converter to HiveMetadata

Stepping through code on who uses `ThriftHiveMetastore`, it looks like this is the only valid codepath: https://fburl.com/pvj8q4fo which gets hit when instantiating [`HiveHadoop2Plugin`](https://fburl.com/biggrep/9vjmci2s) and [`HivePlugin`](https://fburl.com/s6xg9mfw) when we don't pass in the optional `ExtendedHiveMetastore`. Question here is: what does it mean when we do not pass in `ExtendedHiveMetastore` to the plugins? Is it safe to fallback to default `HiveColumnConverter` when `ExtendedHiveMetastore` is missing?

Reply:
TLDR is that it should be safe. I can explain more in person